### PR TITLE
Whfhri-669: Migrate CI from Travis to GH Actions

### PR DESCRIPTION
**Pull request by aramcodz**
Migration of the api-spec Repo CI/build & package processing from TravisCI to Github Actions

### CODE REVIEW

**Comment by aramcodz** [comment]
I have 2 question on this PR. 
1. I have **not yet** Deleted/Removed the `.travis.yml` on this branch for this story? Should I do that now?  Is there any reason for me to hold off on deleting the `.travis.yml` ? 
2. `develop` branch is the correct target for this PR, correct?

**Review by David-N-Perkins — COMMENTED** [review]
You can go ahead and delete the .travis.yml file.
`develop` is the correct target, but you'll need to create similar PR's for support 1.x & 2.x branches.

**Comment by aramcodz** [comment]
> You can go ahead and delete the .travis.yml file.
> `develop` is the correct target, but you'll need to create similar PR's for support 1.x & 2.x branches.

OK - Got it. Thx.

**Review by FRicci618 — APPROVED** [review]

File: .github/workflows/ci-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      
+      - name: set Branch env var
+        if: startsWith(github.ref, 'refs/tags/v') == false
+        run: | 
+          BRANCH_NAME=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          echo "ACTIONS_BRANCH=$BRANCH_NAME" >> $GITHUB_ENV
+
+      - name: Set Tag env var
+        if: startsWith(github.ref, 'refs/tags/v') == true
+        shell: bash
```
[the remark this answers is not in the document]
Comment (aramcodz) [reply .github/workflows/ci-workflow.yaml]: I was following some code example.  
I removed it now - see latest.

File: .github/workflows/ci-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        run: | 
+          TAG_ID=${GITHUB_REF##*/}
```
[the remark this answers is not in the document]
Comment (aramcodz) [reply .github/workflows/ci-workflow.yaml:41]: OK. Done. 
see latest.

File: .github/workflows/ci-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          distribution: 'adopt'
+          cache: 'gradle'
+      
+      - name: run openapi Validate
+        uses: mbowman100/swagger-validator-action@master
+        with:
+          files: |
+            management-api/management.yml
+      
+      - name: set Branch env var
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yaml:32]: Nit -- some capitalization inconsistencies in the step names. (E.g. "run" and "step" are each capitalized in one name and not in another. )

**Review by dan-baxter — APPROVED** [review]

File: .github/workflows/ci-workflow.yaml (hunk 3 above)
Comment (aramcodz) [reply .github/workflows/ci-workflow.yaml]: I have made these more consistent now.  

See latest.

**Review by David-N-Perkins — APPROVED** [review]

**Pull request merged**